pkg/apis/planetscale/v2/vitesscluster_types.go

```diff
@@ -135,6 +135,23 @@ type VitessClusterUpdateStrategy struct {
 	// Default: External
 	// +kubebuilder:validation:Enum=External,Immediate
 	Type *VitessClusterUpdateStrategyType `json:"type,omitempty"`
+
+	// DataVolumeClaimResize selects the strategy chosen when updating the underlying storage
+	// value in the DataVolumeClaimTemplate for a tablet pool's persistent volumes.
+	//
+	// Supported options are:
+	//
+	// - External: Schedule disk size updates on tablet pools and persistent
+	//   volume claims but wait for an external tool to release them by adding
+	//   the 'rollout.planetscale.com/released' annotation.
+	// - Immediate: Release updates to all tablet pools and persistent volume
+	//   claims as soon as the VitessCluster spec is changed. Perform rolling
+	//   restart of one tablet Pod per shard at a time, with automatic planned
+	//   reparents whenever possible to avoid master downtime.
+	//
+	// Default: External
+	// +kubebuilder:validation:Enum=External,Immediate
+	DataVolumeClaimResize *DataVolumeClaimResizeType `json:"dataVolumeClaimResize,omitempty"`
```

Comment:
My feeling is that this shouldn't be separate from the `type` field. If we keep following this pattern, we'll have more toggles than a user knows what to do with. I think it should just all get pushed into `type`.

Reply:
I agree overall too, I was thinking about this... but the interest on this tech debt doesn't seem too bad for the time being, so I was thinking we approach this in a team meeting.

Reply:
My sense is that this is the wrong place to put this if it's an override. Essentially the types are no different at all (it's either immediate or external). I would prefer it being it's own field and just using the existing struct in its entirety. so rather than on `updateStrategy` field, it could be `dataVolumeUpdateStrategy` field. To me this reads as an odd duplication when the it fundamentally represents the exact same two options.

Reply:
Or better yet, have the update strategy live on a child where it's the most relevant to data volumes.

Reply:
> being it's own field and just using the existing struct in its entirety

I was envisioning that the `VitessClusterUpdateStrategy` would eventually have more settings specific to the scope of a Vitess Cluster (like how to update keyspaces in particular), so I don't think we should share the whole struct. It might make sense to share the `VitessClusterUpdateStrategyType` type, but even then I feel like we may need values there that only make sense in certain scopes, like the scope of a whole VitessCluster.

> so rather than on updateStrategy field, it could be dataVolumeUpdateStrategy field

The trade-off there is that if you extend this pattern for other things we update (e.g. keyspaces), the top-level VitessCluster.Spec starts to feel polluted with similar fields (`*UpdateStrategy`) that should instead be grouped:

```yaml
# everything under this tree is assumed to be about update strategy,
# so field names can be cleaner and related concepts are grouped.
updateStrategy:
  # data volume claim update strategy
  dataVolumeClaim: blah
  # keyspace update strategy
  keyspace: blah
  # cell update strategy
  cell: blah
```

This is what we were going for with `dataVolumeClaimResize` but maybe we should generalize it to `dataVolumeClaim` and make it about any updates to the data volume PVC?

> have the update strategy live on a child where it's the most relevant to data volumes

If we were talking about overriding the strategy for a whole subtree of our hierarchy like a shard or a keyspace, then I'd agree with embedding another copy of the struct in the root of the subtree. However, the set of all data volume claims in the cluster is not a subtree; it's a bunch of leaves from all over the tree. What we want here is to affect all PVCs in the cluster, so VitessCluster spec is already the root of the subtree we want to affect.

I do agree overall that this API doesn't feel right, so it's worth spending a bit of time to see if we can come up with something better.

For context, the `updateStrategy` field was meant to mimic the pattern used by StatefulSet's `updateStrategy` and Deployment's `strategy` so users would find it familiar. In that pattern, the `updateStrategy` field is a struct with not only an overall `type` but also subfields to customize settings for the selected type:

```yaml
updateStrategy:
  type: RollingUpdate
  rollingUpdate:
    # when rolling update is enabled, customize how exactly to roll
    foo: bar
```

Perhaps we could double down on copying the k8s API and think of this new field we're adding as a customization of the External strategy type:

```yaml
updateStrategy:
  type: External
  external:
    # Even though our overall strategy type is External,
    # we have customizations to let certain things through.
    allowResourceChanges:
    - storage
    # later we would support these too:
    - cpu
    - memory
```

Reply:
I'm a fan of copying the k8s API, I think it's a nice solution that incorporates Peter's suggestions and gets across the idea that choosing immediate on resources is more like a bypass that doesn't live independently of the `external` option.

Reply:
I like this suggestion - I think it reads much better. I'm on board.